test/integration/02-user-wallet/02-receive-lightning.spec.ts

```diff
@@ -64,4 +70,33 @@ describe("UserWallet - Lightning", () => {
     const { BTC: finalBalance } = await userWallet1.getBalances()
     expect(finalBalance).toBe(initBalance1 + sats)
   })
+
+  it("receives spam invoice", async () => {
+    // amount below MEMO_SPAM threshold
+    const sats = 100
+    const memo = "THIS MIGHT BE SPAM!!!"
+
+    // confirm that transaction should be filtered
+    expect(sats).toBeLessThan(MEMO_SHARING_SATS_THRESHOLD)
+
+    // process spam transaction
+    const invoice = await userWallet1.addInvoice({ value: sats, memo })
+    const hash = getHash(invoice)
+    await pay({ lnd: lndOutside1, request: invoice })
+    expect(await userWallet1.updatePendingInvoice({ hash })).toBeTruthy()
+
+    // check that spam memo is persisted to database
+    const dbTx = await ledger.getTransactionByHash(hash)
+    expect(dbTx.memo).toBe(memo)
+
+    // check that spam memo is filtered from transaction description
+    const txns = await userWallet1.getTransactions()
+    const spamTxn = txns.filter((txn) => txn.hash === hash)[0]
```

Comment:
you can use `txns.find` to avoid `[0]`

Reply:
Didn't know of this method, thanks for the tip :+1: https://github.com/GaloyMoney/galoy/pull/415/commits/9c698c51794249212bc63967ddff5c7f57a85d64